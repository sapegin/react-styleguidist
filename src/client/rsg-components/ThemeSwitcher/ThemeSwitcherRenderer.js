import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import map from 'lodash/map';

const styles = ({ color, fontFamily, fontSize }) => ({
	root: {
		fontFamily: fontFamily.base,
		fontSize: fontSize.base,
		color: color.base,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'space-between',
	},
});

export function ThemeSwitcherRenderer({ classes, themes, currentTheme, onThemeChange }) {
	const handleThemeChange = e => onThemeChange(e.target.value);

	return (
		<label className={classes.root}>
			Theme
			<select value={currentTheme} onBlur={handleThemeChange} onChange={handleThemeChange}>
				{map(themes, ({ id }) => (
					<option key={id}>{id}</option>
				))}
			</select>
		</label>
	);
}

ThemeSwitcherRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	themes: PropTypes.array.isRequired,
	currentTheme: PropTypes.string,
	onThemeChange: PropTypes.func.isRequired,
};

export default Styled(styles)(ThemeSwitcherRenderer);
