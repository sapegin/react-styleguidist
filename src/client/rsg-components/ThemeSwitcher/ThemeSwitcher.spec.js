import React from 'react';
import { ThemeSwitcherRenderer, styles } from './ThemeSwitcherRenderer';

const themes = [
	{
		id: 'dark',
	},
	{
		id: 'light',
	},
];

const currentTheme = 'light';

const props = {
	classes: classes(styles),
	onThemeChange: () => null,
};

describe('ThemeSwitcher', () => {
	it('should render current theme', () => {
		const actual = shallow(
			<ThemeSwitcherRenderer {...props} themes={themes} currentTheme={currentTheme} />
		);

		expect(actual).toMatchSnapshot();
	});
});
