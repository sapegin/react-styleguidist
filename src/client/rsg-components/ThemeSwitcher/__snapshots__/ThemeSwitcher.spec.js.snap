// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher should render current theme 1`] = `
<label
  className="root"
>
  Theme
  <select
    onBlur={[Function]}
    onChange={[Function]}
    value="light"
  >
    <option
      key="dark"
    >
      dark
    </option>
    <option
      key="light"
    >
      light
    </option>
  </select>
</label>
`;
